controllers/controllers/workloads/shared.go

```diff
@@ -38,29 +44,173 @@ func setStatusConditionOnLocalCopy(conditions *[]metav1.Condition, conditionType
 	})
 }
 
-func createSubnamespaceAnchor(ctx context.Context, client client.Client, req ctrl.Request, object client.Object, labels map[string]string) (v1alpha2.SubnamespaceAnchor, error) {
-	anchor := v1alpha2.SubnamespaceAnchor{
+func createOrPatchNamespace(ctx context.Context, client client.Client, log logr.Logger, object client.Object, labels map[string]string) error {
```

Comment:
Back in the CF-for-VMs days when we were using lager, there was sort of a rule that logger should be the first argument in the args list, this helped for better readability. I would suggest having the logger as first argument, the context as second (or vice versa, I can't really make up my mind) and then the rest. Same goes for the rest of the functions in this file

Reply:
We don't have a documented pattern for argument order in the codebase, but the most common order appears to be (ctx, client, logger, ...). It's also a golang convention that ctx should always be first if it's present, just like how error should be last. Perhaps we should have a team discussion on parameter order conventions, but for now we're going to leave this as is.